src/main/groovy/org/labkey/gradle/plugin/FileModule.groovy

```diff
@@ -405,38 +407,69 @@ class FileModule implements Plugin<Project>
                 project.tasks.register("pomFile", PomFile)  {
                     PomFile pFile ->
+                        pFile.description = "Create the pom file for this project's api jar"
                         pFile.group = GroupNames.PUBLISHING
-                        pFile.description = "create the pom file for this project"
-                        pFile.pomProperties = project.lkModule.modProperties
+                        pFile.pomProperties = LabKeyExtension.getApiPomProperties(project)
+                        pFile.isModulePom = false
+                }
+                project.tasks.register("modulePomFile", PomFile)  {
+                    PomFile pFile ->
+                        pFile.description = "Create the pom file for this project's .module file"
+                        pFile.group = GroupNames.PUBLISHING
+                        pFile.pomProperties = LabKeyExtension.getModulePomProperties(project)
+                        pFile.isModulePom = true
                 }
                 project.publishing {
                     publications {
-                        libs(MavenPublication) { pub ->
-                            project.tasks.each {
-                                if (it instanceof Jar &&
-                                        (!it.name.equals("schemasJar") || XmlBeans.isApplicable(project)))
-                                {
-                                    pub.artifact(it)
-                                }
+                        if (project.hasProperty('module'))
+                        {
+                            modules(MavenPublication) { pub ->
+                                    // Use org.labkey.module for module dependency groupIds instead of "org.labkey"
+                                    pub.groupId = LabKeyExtension.MODULE_GROUP
+                                    pub.artifact(project.tasks.module)
+                            }
+                        }
+
+                        if (project.hasProperty('apiJar'))
+                        {
+                            apiLib(MavenPublication) { pub ->
+                                pub.groupId = LabKeyExtension.API_GROUP
+                                pub.artifact(project.tasks.apiJar)
+                            }
+                        }
+                        else if (project.path.equals(BuildUtils.getApiProjectPath(project.gradle))
+                                || project.path.equals(BuildUtils.getInternalProjectPath(project.gradle)))
+                        {
+                            apiLib(MavenPublication) { pub ->
+                                pub.groupId = LabKeyExtension.API_GROUP
+                                pub.artifact(project.tasks.jar)
                             }
                         }
                     }
 
                     if (BuildUtils.shouldPublish(project))
                     {
                         project.artifactoryPublish {
```

Comment:
Can/should we do lazy task configuration here? Something like:
```suggestion
                         project.tasks.named("artifactoryPublish").configure {
```

Reply:
Probably a good idea, but not needed at the moment.  This is a task that only gets configured when the property is set to do publishing, so not a lot of savings in any case.